flink-connector/src/main/java/org/apache/flink/connector/delta/sink/commiter/DeltaGlobalCommitter.java

```diff
@@ -0,0 +1,134 @@
+package org.apache.flink.connector.delta.sink.commiter;
+
+import io.delta.standalone.CommitResult;
+import io.delta.standalone.DeltaLog;
+import io.delta.standalone.Operation;
+import io.delta.standalone.OptimisticTransaction;
+import io.delta.standalone.actions.Action;
+import io.delta.standalone.actions.AddFile;
+import io.delta.standalone.actions.Metadata;
+import org.apache.flink.api.connector.sink.GlobalCommitter;
+import org.apache.flink.connector.delta.sink.committables.DeltaCommittable;
+import org.apache.flink.connector.delta.sink.committables.DeltaGlobalCommittable;
+import org.apache.flink.connector.delta.sink.writer.DeltaPendingFile;
+import org.apache.flink.core.fs.Path;
+import org.apache.hadoop.conf.Configuration;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DeltaGlobalCommitter implements GlobalCommitter<DeltaCommittable, DeltaGlobalCommittable> {
+
+    private final String engineInfo = "flink.1.12"; // parametrize
+
+    private final Configuration conf;
+
+    private final Path basePath;
+
+    public DeltaGlobalCommitter(Configuration conf,
+                                Path basePath) {
+        this.conf = conf;
+        this.basePath = basePath;
+    }
+
+
+    @Override
+    public List<DeltaGlobalCommittable> filterRecoveredCommittables(List<DeltaGlobalCommittable> globalCommittables) throws IOException {
+        return globalCommittables;
+    }
+
+    @Override
+    public DeltaGlobalCommittable combine(List<DeltaCommittable> committables) throws IOException {
+        List<DeltaPendingFile> pendingFiles = new ArrayList<>();
+        for (DeltaCommittable committable : committables) {
+            if (committable.hasDeltaPendingFile()) {
+                assert committable.getDeltaPendingFile() != null;
+                assert committable.getDeltaPendingFile().getFileName() != null;
+                pendingFiles.add(committable.getDeltaPendingFile());
+            }
+        }
+        return new DeltaGlobalCommittable(pendingFiles);
+    }
+
+    @Override
+    public List<DeltaGlobalCommittable> commit(List<DeltaGlobalCommittable> globalCommittables) throws IOException {
+        if (!globalCommittables.isEmpty()) {
+            List<Action> addFileActions = prepareAddFileActions(globalCommittables);
+
+            Operation operation = prepareOperation(globalCommittables, addFileActions.size());
+            DeltaLog deltaLog = DeltaLog.forTable(conf, basePath.getPath());
+
+            Metadata metadata = deltaLog.snapshot().getMetadata();
+
+            OptimisticTransaction transaction = deltaLog.startTransaction();
+
+            CommitResult result = transaction.commit(addFileActions, operation, this.engineInfo);
```

Comment:
transaction.commit can throw, so we might want to catch that and handle the exception accordingly. for example, what is this function supposed to return? the DeltaGlobalCommittables that were not successfully committed?